File: src/dotnet/FaxCore.ApiSamples/Samples/Administration/Domain/CreateDomain.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.Administration.Domain
{
    internal static class CreateDomain
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            if (!config.ParentDomainID.HasValue || string.IsNullOrWhiteSpace(config.DomainName))
            {
                throw new InvalidOperationException("Set FAXCORE_PARENT_DOMAIN_ID and FAXCORE_DOMAIN_NAME before running admin.domain.create.");
            }

            var request = new
            {
                parentDomainID = config.ParentDomainID.Value,
                domainName = config.DomainName,
                description = config.DomainDescription,
                isActive = config.DomainActive
            };

            return client.PostAsync("/api/domain", request);
        }
    }
}

File: src/javascript/administration/domain/create-domain.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const parentDomainID = process.env.FAXCORE_PARENT_DOMAIN_ID;
const domainName = process.env.FAXCORE_DOMAIN_NAME;

if (!parentDomainID || !domainName) {
  throw new Error('FAXCORE_PARENT_DOMAIN_ID and FAXCORE_DOMAIN_NAME are required.');
}

const client = new FaxCoreClient();

const result = await client.post('/api/domain', {
  parentDomainID: Number(parentDomainID),
  domainName,
  description: process.env.FAXCORE_DOMAIN_DESCRIPTION || '',
  isActive: (process.env.FAXCORE_DOMAIN_ACTIVE || 'true').toLowerCase() !== 'false'
});

printResult('Create domain', result);

